tools/lambda-promtail/main.tf

```diff
@@ -88,4 +95,26 @@ resource "aws_cloudwatch_log_subscription_filter" "lambdafunction_logfilter" {
   # required but can be empty string
   filter_pattern = ""
   depends_on     = [aws_iam_role_policy.logs]
-}
\ No newline at end of file
+}
+
+resource "aws_lambda_permission" "allow-s3-invoke-lambda-promtail" {
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.lambda_promtail.arn
+  principal     = "s3.amazonaws.com"
+  source_arn    = "arn:aws:s3:::${var.bucket_names[count.index]}"
+  count         = length(var.bucket_names)
+}
+
+resource "aws_s3_bucket_notification" "push-to-lambda-promtail" {
+  bucket = var.bucket_names[count.index]
```

Comment:
use a `for_each` here

Reply:
I thought about that but it was previously configured this way so I followed that

Reply:
if `for_each` is the current idiomatic approach in terraform then :+1: from me, it's been a while since I wrote much terraform